### PR TITLE
[TS] LPS-201603

### PR DESCRIPTION
**Pull request by ChrisKian**
https://liferay.atlassian.net/browse/LPS-201603

Please let me know if you have any questions, thanks!

### CODE REVIEW

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

File: modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/internal/upgrade/v1_4_0/ListTypeUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	protected void doUpgrade() throws Exception {
+		try (LoggingTimer loggingTimer = new LoggingTimer();
+			PreparedStatement preparedStatement = connection.prepareStatement(
+				"select listTypeId, type_ from ListType where name IS NULL " +
+					"OR name = ''");
+			ResultSet resultSet = preparedStatement.executeQuery()) {
+
+			while (resultSet.next()) {
+				long listTypeId = resultSet.getLong("listTypeId");
+				String type = resultSet.getString("type_");
```
Comment (smottal) [inline modules/apps/list-type/list-type-service/src/main/java/com/liferay/list/type/internal/upgrade/v1_4_0/ListTypeUpgradeProcess.java:30]: Can we add the types also as condition in the sql? like 
```... where (name IS NULL or name = '') and (type_ = '" + ListTypeConstants.CONTACT_PREFIX + "' or type_ = '"+ ListTypeConstants.CONTACT_SUFFIX +"')``` ?

In this way we will loop on fewer items

Or probably is better to perform directly two updates without the select and the loop like
```update Contact_ set prefixListTypeId = 0 where prefixListTypeId in (select listTypeId from ListType where (name IS NULL or name = '') and (type_ = '"+ListTypeConstants.CONTACT_PREFIX+"'))```
and 
```update Contact_ set suffixListTypeId = 0 where suffixListTypeId in (select listTypeId from ListType where (name IS NULL or name = '') and (type_ = '"+ListTypeConstants. CONTACT_SUFFIX +"'))```

What do you think?

File: modules/apps/list-type/list-type-service/bnd.bnd [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Bundle-Name: Liferay List Type Service
 Bundle-SymbolicName: com.liferay.list.type.service
 Bundle-Version: 1.0.41
-Liferay-Require-SchemaVersion: 1.3.0
+Liferay-Require-SchemaVersion: 1.4.0
```
Comment (smottal) [inline modules/apps/list-type/list-type-service/bnd.bnd:4]: Is it correct that updating only data we need to increase the minor version and not only the patch one? like 1.3.1

**Review by smottal — CHANGES_REQUESTED** [review]
Left some comments on the PR. Thx

**Comment by ChrisKian** [comment]
Hey @smottal,
I've made and tested the requested changes, and everything works as intended.  Please let me know if you have any questions, thanks!

**Comment by ChrisKian** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
ci:test:relevant

**Review by smottal — APPROVED** [review]
LGTM! Thx.

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:um

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:stable

**Comment by smottal** [comment]
Reset here for rebase: https://github.com/liferay-commerce/liferay-portal/pull/4498

**Pull request closed without merging**